### PR TITLE
Offline-First™

### PR DESCRIPTION
**Pull request by ndorfin**
Another First for Abletech, in the series of Firsts. e.g. Mobile-First™.

This PR adds a Service Worker that does:

- offline-pre-caching, and then also…
- caching requests for images etc, that match the target domain.

I've added images to the pre-cache ignore list, because with them included the initial page hit was over 3.7MB. If we cache the HTML, CSS and JS, the request is still under 1MB. #winning

Winning

### CODE REVIEW

File: source/service-worker.js.erb [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+				if (response) { // if cache is available, respond with cache
+					<%= "console.log('responding with cache : ' + e.request.url);" if env_name != 'production' %>
+					return response;
+				}
+				// if there are no cache, try fetching request
+				<%= "console.log('file is not cached, fetching and caching: ' + e.request.url);" if env_name != 'production' %>
+				return fetch(e.request.clone()).then(function(response) {
+					<%= "console.log('  Response for %s from network is: %O', e.request.url, response);" if env_name != 'production' %>
+					var matchesDomain = e.request.url.indexOf('<%= url_root %>') > -1;
+					console.log('matchesDomain', matchesDomain);
```
Comment (jamesbowles) [inline source/service-worker.js.erb]: @ndorfin Do you want to log this out in prod mode?

**Comment by jamesbowles** [comment]
high five!

File: source/service-worker.js.erb (hunk 1 above)
Comment (ndorfin) [inline source/service-worker.js.erb]: Ahh, nope. Will fix that.

**Comment by ndorfin** [comment]
@jamesbowles Wait, who is Iceman?

**Pull request merged**